src/components/custom-field-input-date/custom-field-input-date.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import calendarSvg from '../../svgs/icon-calendar-fill.svg';
+import Icon from '../icon/icon.jsx';
+import CustomFieldInputText from '../custom-field-input-text/custom-field-input-text.jsx';
+import { convertToFormat, validDate } from './format/format-date.js';
+
+const isValidInput = (value) => {
+  if (value === '' || value === undefined) {
+    return true;
+  }
+
+  return validDate(value);
+};
+
+const isValueValid = (value, error, isInputValid = false) => {
+  if (error) {
+    return false;
+  }
+
+  if (!isInputValid) {
+    return false;
+  }
+
+  return isValidInput(value);
+};
+
+export default function CustomFieldInputDate(props) {
+  const inputRef = useRef(null);
+  const [isEditing, setIsEditing] = useState(true);
+  const [isValid, setIsValid] = useState(isValueValid(props.value, props.error, true));
+  const [isFocused, setIsFocused] = useState(false);
+
+  useEffect(() => {
+    const isInputValid = inputRef.current.validity.valid;
+    const valid = isValueValid(props.value, props.error, isInputValid);
+    setIsValid(valid);
+
+    if (!valid) {
+      setIsEditing(true);
+    } else {
+      setIsEditing(false);
+    }
+  }, [inputRef.current]);
```

Comment:
We should also probably look at `props.value` and `props.error`.

Reply:
We might want to also scope it down to: `inputRef.current.validity.valid` but I am not sure. Is the idea that we don't want to re-run this method when the element reference does not change? Or when the validity of the element does not change?

Reply:
This array is things to watch and re-run when they _do_ change, so we definitely want to know if the `inputRef.current` changes.